chain/x/zoracle/internal/keeper/request_test.go

```diff
@@ -4,49 +4,150 @@ import (
 	"testing"
 
 	"github.com/bandprotocol/d3n/chain/x/zoracle/internal/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
 )
 
+func newDefaultRequest() types.Request {
+	return types.NewRequest(
+		1,
+		[]byte("calldata"),
+		[]sdk.ValAddress{sdk.ValAddress([]byte("validator1")), sdk.ValAddress([]byte("validator2"))},
+		2,
+		0,
+		1581503227,
+		100,
+	)
+}
+
 func TestGetterSetterRequest(t *testing.T) {
 	ctx, keeper := CreateTestInput(t, false)
 
 	_, err := keeper.GetRequest(ctx, 1)
 	require.NotNil(t, err)
 
-	request := types.NewRequest([]byte("CodeHash"), []byte("params"), 10)
+	request := newDefaultRequest()
+
+	keeper.SetRequest(ctx, 1, request)
+	actualRequest, err := keeper.GetRequest(ctx, 1)
+	require.Nil(t, err)
+	require.Equal(t, request, actualRequest)
+}
+
+// TestAddSubmitValidator tests keeper can add valid validator to request
+func TestAddSubmitValidator(t *testing.T) {
+	ctx, keeper := CreateTestInput(t, false)
+	request := newDefaultRequest()
+
+	keeper.SetRequest(ctx, 1, request)
+
+	err := keeper.AddSubmitValidator(ctx, 1, sdk.ValAddress([]byte("validator1")))
+	require.Nil(t, err)
+
+	actualRequest, err := keeper.GetRequest(ctx, 1)
+	request.SubmittedValidatorList = []sdk.ValAddress{sdk.ValAddress([]byte("validator1"))}
+	require.Nil(t, err)
+	require.Equal(t, request, actualRequest)
+}
+
+// TestAddSubmitValidatorOnInvalidRequest tests keeper must return if add on invalid request
+func TestAddSubmitValidatorOnInvalidRequest(t *testing.T) {
+	ctx, keeper := CreateTestInput(t, false)
+	request := newDefaultRequest()
+
+	keeper.SetRequest(ctx, 1, request)
+	err := keeper.AddSubmitValidator(ctx, 2, sdk.ValAddress([]byte("validator1")))
+	require.Equal(t, types.CodeRequestNotFound, err.Code())
+}
+
+// TestAddInvalidSubmitValidator tests keeper return error if try to add new validator that doesn't contain in list.
+func TestAddInvalidSubmitValidator(t *testing.T) {
+	ctx, keeper := CreateTestInput(t, false)
+	request := newDefaultRequest()
 
 	keeper.SetRequest(ctx, 1, request)
+
+	err := keeper.AddSubmitValidator(ctx, 1, sdk.ValAddress([]byte("validator3")))
+	require.Equal(t, types.CodeInvalidValidator, err.Code())
+
 	actualRequest, err := keeper.GetRequest(ctx, 1)
```

Comment:
check `validator3` is not in `actualRequest` ?

Reply:
Test equal in line 75 already